networksurvey/src/main/java/com/craxiom/networksurvey/fragments/CodeScannerFragment.java

```diff
@@ -0,0 +1,98 @@
+package com.craxiom.networksurvey.fragments;
+
+import android.app.Activity;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.navigation.Navigation;
+
+import com.budiyev.android.codescanner.CodeScanner;
+import com.budiyev.android.codescanner.CodeScannerView;
+import com.craxiom.networksurvey.R;
+import com.craxiom.networksurvey.fragments.model.ScannedSettings;
+import com.google.gson.Gson;
+
+/**
+ * Fragment responsible for QR code scanning. Leverages an open source code scanning library from
+ * Yuriy Budiev.
+ *
+ * @since 1.7.0
+ */
+public class CodeScannerFragment extends Fragment
+{
+    private CodeScanner codeScanner;
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
```

Comment:
One thing I noticed is that the logging toggle buttons in the toolbar show up when this fragment is opened. I probably should remove those buttons for all the fragments from the nav drawer, but you should probably remove it for this fragment. It is pretty easy to do, I think the call is something like `setHasOptionsMenu(false);` in the onCreate() method. If that does not work you might need to change that to `setHasOptionsMenu(true);` and then override this method with something like...

```
    @Override
    public void onPrepareOptionsMenu(@NonNull Menu menu)
    {
        menu.findItem(R.id.some_menu_action).setVisible(false);
        super.onPrepareOptionsMenu(menu);
    }
```